src/Npgsql/NpgsqlRawCopyStream.cs

```diff
@@ -89,49 +89,100 @@ internal NpgsqlRawCopyStream(NpgsqlConnector connector, string copyCommand)
 
         #region Write
 
-        public override void Write(byte[] buffer, int offset, int count) => Write(buffer, offset, count, false).GetAwaiter().GetResult();
+        public override void Write(byte[] buffer, int offset, int count)
+            => Write(new ReadOnlySpan<byte>(buffer, offset, count));
 
         public override Task WriteAsync(byte[] buffer, int offset, int count, CancellationToken cancellationToken)
-        {
-            if (cancellationToken.IsCancellationRequested)
-                return Task.FromCanceled(cancellationToken);
-            using (NoSynchronizationContextScope.Enter())
-                return Write(buffer, offset, count, true);
-        }
+            => WriteAsync(new Memory<byte>(buffer, offset, count), cancellationToken).AsTask();
 
-        async Task Write(byte[] buffer, int offset, int count, bool async)
+#if !NET461 && !NETSTANDARD2_0
+        public override void Write(ReadOnlySpan<byte> buffer)
```

Comment:
We can probably push this (and its Memory overload) down into NpgsqlWriteBuffer), and possibly reuse elsewhere - but it doesn't have to be in this PR.

Reply:
Not sure to understand... I prefer to keep it like this for now.